packages/kit/src/utils/url.js

```diff
@@ -6,39 +6,20 @@ import { BROWSER } from 'esm-env';
  */
 export const SCHEME = /^[a-z][a-z\d+\-.]+:/i;
 
-const absolute = /^([a-z]+:)?\/?\//;
+const internal = new URL('sveltekit-internal://');
 
 /**
  * @param {string} base
  * @param {string} path
  */
 export function resolve(base, path) {
-	if (SCHEME.test(path)) return path;
-	if (path[0] === '#') return base + path;
-	if (path === '') return base;
+	// special case
+	if (path[0] === '/' && path[1] === '/') return path;
```

Comment:
what's this special case about? Is this some extra thing we need to mention in the docs? What does this mean for `''`?

Reply:
protocol-relative URLs should remain protocol-relative, but the `new URL(...)` technique means we lose that information. so we need to special-case it